src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -835,6 +845,11 @@ private async Task<bool> SendBreakpointsOfMethodUpdated(SessionId sessionId, Exe
                 {
                     await SetBreakpoint(sessionId, context.store, req, true, token);
                 }
+                //trying to reset breakpoints that were not bound but are in the same file of a method update
+                if (req.Method == null && method.Info.SourceName.Equals(req.File))
+                {
+                    await SetBreakpoint(sessionId, context.store, req, true, token);
+                }
```

Comment:
I think the whole loop can become something like
```suggestion
             foreach (var req in context.BreakpointRequests.Values)
            {
                if (req.TryResolve(method.Source))
                {
                    await SetBreakpoint(sessionId, context.store, req, true, token);
                }
            }
```

Reply:
We could refactor the setbreakpoint logic to be more efficient after that by only updating on method if needed

Reply:
Done.